### PR TITLE
1.7

### PR DESCRIPTION
**Pull request by korotkoves**
multiple-sitl project uses this modifications. Minimal support needs this commit: https://github.com/PX4/sitl_gazebo/commit/5550aaaac22b9b84e604a378039d9caac5d8c274

Pull request is for multiple sitl support in px4 project (https://github.com/PX4/Firmware/issues/8512)

### CODE REVIEW

File: include/common.h [hunk 1]
```diff
@@ -49,6 +50,36 @@ bool getSdfParam(sdf::ElementPtr sdf, const std::string& name, T& param, const T
   return false;
 }
 
+template <typename T>
```
Comment (TSC21) [inline include/common.h:53]: Can you please put this on a separate PR?

File: include/gazebo_gps_plugin.h [hunk 2]
```diff
@@ -102,7 +102,7 @@ class GAZEBO_VISIBLE GpsPlugin : public ModelPlugin
   static constexpr const double earth_radius = 6353000.0;      // meters
 
   // gps delay related
-  static constexpr double gps_update_interval_ = 0.2; // 5hz
+  double gps_update_interval_ = 0.2; // 5hz
```
Comment (TSC21) [inline include/gazebo_gps_plugin.h:105]: Any reason you would want to change the GPS rate? I mean the common rate for GPS is 5-10hz.

File: include/gazebo_imu_plugin.h [hunk 3]
```diff
@@ -134,6 +134,7 @@ class GazeboImuPlugin : public ModelPlugin {
   event::ConnectionPtr updateConnection_;
 
   common::Time last_time_;
+  int imu_rate_;
```
Comment (TSC21) [inline include/gazebo_imu_plugin.h:137]: This is already address in #157. It will be merged first.

File: include/gazebo_mavlink_interface.h [hunk 4]
```diff
@@ -236,6 +235,7 @@ class GazeboMavlinkInterface : public ModelPlugin {
   int _fd;
   struct sockaddr_in _myaddr;     ///< The locally bound address
   struct sockaddr_in _srcaddr;    ///< SITL instance
+  struct sockaddr_in _hil_gps_addr;
```
Comment (TSC21) [inline include/gazebo_mavlink_interface.h:238]: what's the purpose of this socket address?

File: include/gazebo_mavlink_interface.h [hunk 5]
```diff
@@ -249,5 +249,7 @@ class GazeboMavlinkInterface : public ModelPlugin {
 
   in_addr_t mavlink_addr_;
   int mavlink_udp_port_;
+
+  bool hil_state_;
```
Comment (TSC21) [inline include/gazebo_mavlink_interface.h:253]: Already addressed in #157.

**Review by TSC21 — CHANGES_REQUESTED** [review]
@korotkoves can you please clarify some of the questions I made on comments? Also, some changes are already going to be pushed on #157 soon.

File: include/gazebo_gps_plugin.h (hunk 2 above)
Comment (korotkoves) [reply include/gazebo_gps_plugin.h:105]: to emulate gps receivers with rate more then 5hz

File: include/common.h (hunk 1 above)
Comment (korotkoves) [reply include/common.h:53]: ок

File: include/gazebo_mavlink_interface.h (hunk 4 above)
Comment (korotkoves) [reply include/gazebo_mavlink_interface.h:238]: to emulate GPS receiver directly connected to on-board processor

Comment (TSC21) [reply include/gazebo_mavlink_interface.h:238]: This needs a correspondent comment explaining this.

**Comment by TSC21** [comment]
@korotkoves can you please rebase this PR and update it accordingly? Thanks

**Comment by korotkoves** [comment]
redone with https://github.com/PX4/sitl_gazebo/pull/199

**Comment by LorenzMeier** [comment]
@TSC21 Could you rebase and retest?

**Comment by TSC21** [comment]
Superseded by #199
